### ISSUE TITLE
Bug: When Subscription has fragments but is NOT named Prisma returns null right away and closes subscription

### ISSUE THREAD

**Issue by ntziolis** [prisma/prisma#2942]
**Summary**
When using fragments + NOT naming the subscription Prisma (not app endpoint) returns `{ data: null }` straight away  + closes the connection (no spinner in playground).

**To Reproduce**

1. Feel free to use my endpoint: https://eu1.prisma.sh/public-silvermole-100/subscription-fragment/dev
or
Create new service in prisma with the following schema:
```
type Post {
  id: ID! @unique
}
```
2. Execute the below in playground against prisma endpoint
```
fragment PostFragment on Post{
  id
}
subscription {post{node{  ...PostFragment  }}}
```
3. See error
Returns `{ data: null }` straight away  + closes the connection (no spinner in playground)

**Expected behavior**
Subscription not be closed (spinning wheel in playground). Data should be returned when there is an event the subscription matches.

**Screenshots**

**Versions (please complete the following information):**
 - Prisma Server: 1.15-beta-1 (current dev servers)

**Context**
When either giving the subscription a name or removing the fragments it works just fine (spinning wheel in playground):
```
fragment PostFragment on Post{
  id
}
subscription SomeName{post{node{  ...PostFragment  }}}
```
or
```
subscription {post{node{  id  }}}
```

**Comment by marktani** [comment]
This seems to be a bug in the Playground. Instead of opening a websocket connection, the Playground seems to send a query, which returns the `null` data:

```
curl 'https://eu1.prisma.sh/public-silvermole-100/subscription-fragment/dev' -H 'origin: https://eu1.prisma.sh' -H 'accept-encoding: gzip, deflate, br' -H 'accept-language: en-US,en;q=0.9,de;q=0.8,es;q=0.7' -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36' -H 'content-type: application/json' -H 'accept: */*' -H 'referer: https://eu1.prisma.sh/public-silvermole-100/subscription-fragment/dev' -H 'authority: eu1.prisma.sh' --data-binary '{"operationName":null,"variables":{},"query":"fragment PostFragment on Post {\n  id\n}\n\nsubscription {\n  post {\n    node {\n      ...PostFragment\n    }\n  }\n}\n"}' --compressed
```

results in 

```
{
  "data" : {
    "post" : null
  }
}
```

As you can see, the endpoint is `https://eu1.prisma.sh/public-silvermole-100/subscription-fragment/dev` and not `wss://...`, so what the server returns for this is undefined.

I'll move this to the `graphql-playground` repository.

**Comment by marktani** [comment]
Issue moved to [prisma/graphql-playground #846](https://github.com/prisma/graphql-playground/issues/846) via [**ZenHub**](https://www.zenhub.com/)

**Comment by mcmar** [comment]
@marktani This isn't a bug with `graphql-playground`. I'm inspecting the traffic and `graphql-playground` is correctly sending it over a websocket. I can even see the individual websocket frames.
This is a bug with `prisma-binding` and I can confirm that because this bug is not present when I send a subscription directly to my prisma backend. However, it is present when I send the subscription with `prisma-binding`. I can even see that `graphql-playground` sends it as a websocket, but when I enable `debug` on `prisma-binding`, it logs the following:
```
Request to http://hash.us-west-2.elb.amazonaws.com/vibely/prod:
query:
subscription ($_v0_where: ChatRoomMessageSubscriptionWhereInput) {
  chatRoomMessage(where: $_v0_where) {
    node {
      ...Message
    }
  }
}

fragment Message on ChatRoomMessage {
  id
}
operationName: null
variables:
{
  "_v0_where": {
    "mutation_in": [
      "CREATED"
    ],
    "node": {
      "chatRoom": {
        "id": "REDACTED"
      }
    }
  }
}
```

Screenshots included for proof:
<img width="1440" alt="screen shot 2018-11-13 at 6 29 01 pm" src="https://user-images.githubusercontent.com/8251430/48456010-41248300-e772-11e8-905a-0f248ef6937a.png">

I hope that this evidence is sufficient to prove my case. I think we should close the issue with `graphql-playground` because no such issue exists and open a new issue with `prisma-binding` so we can fix this.

**Comment by mcmar** [comment]
TL;DR for future readers, upgrade your version of `prisma-binding` to the latest

**Comment by yodaheis** [comment]
Still facing the same issue, even after upgrading to prisma-binding@2.1.6
Tried from latest version of GraphQL Playground electron app + Browser. Issue remains.
![subscription_fragment_playground-issue](https://user-images.githubusercontent.com/22661681/48605592-33097900-e9a3-11e8-8f97-d23f77a83910.jpg)

**Comment by dunatron** [comment]
so I am on "prisma-binding": "^2.3.16" and still having the same issue. Not a huge deal to not use fragments on the client app with subscriptions but then you have to keep both the fragment and the subscription version up to date which is annoying at best
